<a href="https://colab.research.google.com/github/VidishaPonnappa/MachineLearning/blob/main/Project_18.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#### Goals:

1. Last Digit Ultimate

2. Pentagonal NUmber


---

#### 1. Last Digit Ultimate

Your job is to create a function, that takes 3 numbers: `num1, num2, num3` and returns `True` if:  (last digit of `num1`) * (the last digit of `num2`) = the last digit of `num3`.

In [ ]:
# Write your solution here:
def lastdig(a,b,c):
  sa,sb,sc=str(a),str(b),str(c)
  la,lb,lc=sa[-1],sb[-1],sc[-1]
  nla,nlb,nlc=int(la),int(lb),int(lc)
  nprod=nla*nlb
  nprod=list(nprod)
  print(nprod)
  if nprod[-1]==nlc:
    return True
  else:
    return False
lastdig(25,21,124)

---

#### 2. Pentagonal Number

Given a number $num$ as a positive integer, calculate how many dots exist in a pentagonal shape around the centre dot on the $Nth$ iteration.

In the image below you can see in the first iteration there is only a single dot. On the second, there are 6 dots. On the third, there are 16 dots, and on the fourth there are 31 dots.

<img src='https://student-datasets-bucket.s3.ap-south-1.amazonaws.com/images/pengatonal-number.png' width=500>


In [ ]:
1,6,16,31,51
1+5(0),5(1)+1,5(2)+6

In [ ]:
# Write your solution here
def pent(n):
  num,l=1,[]
  for i in range(0,n):
    num+=(5*i)
    l.append(num)
  print('In series of {} terms, last term is {}.'.format(n,l[-1]))

n=int(input('Enter the number of terms: '))
pent(n)

Enter the number of terms: 9
In series of 9 terms, last term is 181.


**Q**: How many dots are there in the 9th iteration?

**A**: 181

---